### PR TITLE
chore: slurm proxy support

### PR DESCRIPTION
**Pull request by stoksc**
## Description
This change allows containers which would like proxies in slurm to advertise their proxy address to the master. When this address is advertised, the proxy is registered in the master and endpoints concerning proxies will list addresses for the container.

## Test Plan
- [x] test ancillary rendezvous stuff on a slurm
- [x] test proxy stuff on casablanca

## Commentary (optional)
There are a million ways to write this code. I just went with the simplest I could find for now.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/task/allocation.go [hunk 1]
```diff
@@ -66,6 +66,8 @@ type (
 		idleTimeoutWatcher *IdleTimeoutWatcher
 		// proxy state
 		proxies []string
+		// proxyAddress is provided by determined.exec.prep_container if the RM doesn't provide it.
+		proxyAddress *string
```
Comment (ioga) [inline master/internal/task/allocation.go:70]: we need to persist this field to the db [so we are able to load it back on master restart].

Comment (stoksc) [reply master/internal/task/allocation.go:70]: ok, I suppose I need to let you land your pr then.

Comment (stoksc) [reply master/internal/task/allocation.go:69]: well, I'll hold off until we do 'slurm fault tolerance' or something. is that OK?

Comment (ioga) [reply master/internal/task/allocation.go:69]: no problem

**Review by ioga — APPROVED** [review]

**Pull request merged**